numba/runtime/nrt.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import print_function, absolute_import, division
+
+from . import _nrt_python as _nrt
+from . import atomicops
+
+
+class Runtime(object):
+    def __init__(self):
+        # Compile atomic operations
+        compiled = atomicops.compile_atomic_ops()
+        self._mcjit, self._ptr_inc, self._ptr_dec, self._ptr_cas = compiled
+        # Install atomic ops to NRT
+        _nrt.memsys_set_atomic_inc_dec(self._ptr_inc, self._ptr_dec)
+        _nrt.memsys_set_atomic_cas(self._ptr_cas)
+
+    def __del__(self):
+        _nrt.memsys_shutdown()
+
+    def meminfo_new(self, data, pyobj):
+        mi = _nrt.meminfo_new(data, pyobj)
+        return MemInfo(mi)
+
+    def meminfo_alloc(self, size, safe=False):
```

Comment:
I added this during debugging.  I wanted it to mean anything that makes allocation safer.  All I could think of is zero-filling after malloc() and before free().  It helped track down premature deallocation.  I will document what it does.  We can enhance it to do more.  I don't expect it to be enabled for optimized run.